Support an eventual phase for the blueprint actions (#1297)
kanisterio/kanister: docs/tutorial.rst

If you re-execute this Kanister Action, you'll be able to see the Artifact in the
ActionSet status.

> If you have the use case of ``DeferPhase``, below is how you can set the output artifact
> from the output that is being generated from ``DeferPhase``

Reviewer: ```suggestion
If you use a ``DeferPhase``, you can set the output artifact
from the output generated from ``DeferPhase`` as shown below.
```

    namespace: kanister
  actions:
    backup:
      configMapNames:
      - location
      secretNames:
      - aws
      outputArtifacts:
        timeLog:
          keyValue:
            path: '{{ .ConfigMaps.location.Data.path }}/time-log/'
        deferPhaseArt:
          keyValue:
            time: "{{ .DeferPhase.Output.bkpCompletedTime }}"
      phases:
        - func: KubeExec
          name: backupToS3
          args:
            namespace: "{{ .Deployment.Namespace }}"
            pod: "{{ index .Deployment.Pods 0 }}"
            container: test-container
            command:
              - sh
              - -c
              - |
                echo "Main Phase"
      deferPhase:
        func: KubeTask
        name: saveBackupTime
        args:
          image: ghcr.io/kanisterio/mysql-sidecar:0.74.0
          namespace: "{{ .Deployment.Namespace }}"
          command:
          - bash
          - -o
          - errexit
          - -o
          - pipefail
          - -c
          - |
            echo "DeferPhase"
            kando output bkpCompletedTime "10Minutes"
  EOF

Output from the previous phases can also be used in the ``DeferPhase`` like it

> is used in normal scenarios.

Author: the way we set output artifacts for `DeferPhase` is different from how we set it for normal phases. That's the reason I thought its better to have that here.
But I think I can specify just `DeferPhase` in first blueprint section as well. EDYT?
